src/operator/nn/mkldnn/mkldnn_base.cc

```diff
@@ -77,26 +77,35 @@ mkldnn::memory *TmpMemMgr::Alloc(const mkldnn::memory::primitive_desc &pd) {
   }
 }
 
-mkldnn_output_t CreateMKLDNNMem(const NDArray &arr,
-                                const mkldnn::memory::primitive_desc &desc,
-                                OpReqType req) {
+bool CanWriteTo(const NDArray &out_arr,
+                const NDArray* in_arr,
+                const mkldnn::memory::primitive_desc &desc) {
+  bool add_same = in_arr->GetMKLDNNData()->get_data_handle() ==
+      out_arr.GetMKLDNNData()->get_data_handle();
+  bool pdesc_same = out_arr.GetMKLDNNData()->get_primitive_desc() == desc;
```

Comment:
Is `desc` here same with `in_arr.GetMKLDNNData()->get_primitive_desc()`?

Reply:
not necessarily. it is the desc passed directly to CreateMKLDNNMem. and it could be the desc provided by the the operator.

``
mkldnn::sum::primitive_desc pdesc(scales, in_pds);
auto mem = CreateMKLDNNMem(out_data, pdesc.dst_primitive_desc(), req, &inputs[0]);
``